src/index.ts

```diff
@@ -212,6 +217,40 @@ const logoPlugin: JupyterFrontEndPlugin<void> = {
   }
 };
 
-const plugins = [commandsPlugin, logoPlugin];
+const jhubAppsPlugin: JupyterFrontEndPlugin<void> = {
+  id: 'jupyterlab-nebari-mode:jhub-apps-integration',
+  description: 'Deploys an application with (almost) one click.',
+  autoStart: true,
+  requires: [],
+  activate: async (app: JupyterFrontEnd) => {
+    const openURL = (url: string) => {
+      try {
+        window.open(url, '_blank', 'noopener,noreferrer');
+      } catch (error) {
+        console.warn('Error opening URL:', url, error);
+      }
+    };
+
+    app.commands.addCommand(CommandIDs.deployApp, {
+      execute: async () => {
+        const currentWidget = app.shell.currentWidget;
+        const currentNotebookPath =
+          currentWidget && currentWidget instanceof DocumentWidget
+            ? currentWidget.context.path
+            : '';
+        const deployUrl = `/services/japps/create-app?filepath=${encodeURIComponent(currentNotebookPath)}`;
+
+        await openURL(deployUrl);
+      },
+      label: () => 'Deploy App',
+      icon: () => {
+        // Add logic to only display the icon if the command is not in the main menu
```

Comment:
Is there a way to do this i.e. display the icon only except for the main menu (see the screenshots and compare them to the Figma diagrams)?

Reply:
This method takes an argument that can be used to display this icon or not - I don't remember all attributes of top of my head but it should have something like `isToolbar` and the underlying event.

Reply:
Ok, I've just added a new argument (`origin`) to the `schema` JSON that controls this, I couldn't figure out how to get something like `isToolbar` working.